test/server/charset-test.js

```diff
@@ -0,0 +1,329 @@
+                res.set('content-type', 'application/javascript');
+                res.end(iconv.encode(scriptSrc, charsetUrl));
+            });
+
+        app
+            .get('/other-resource/:resourceType/:charsetType', function (req, res) {
+                var contentType;
+                var src;
+
+                switch (req.params.resourceType) {
+                    case 'stylesheet':
+                        contentType = 'text/css';
+                        src         = stylesheetSrc;
+                        break;
+                    case 'manifest':
+                        contentType = 'text/cache-manifest';
+                        src         = manifestSrc;
+                        break;
+                }
+
+                switch (req.params.charsetType) {
+                    case 'bom':
+                        res.set('content-type', contentType + '; charset=' + charsetContentType);
+                        res.end(iconv.encode(src, charsetBOM, { addBOM: true }));
+                        break;
+                    case 'content-type':
+                        res.set('content-type', contentType + '; charset=' + charsetContentType);
+                        res.end(iconv.encode(src, charsetContentType));
+                        break;
+                    default:
+                        res.set('content-type', contentType);
+                        res.end(iconv.encode(src, charsetDefault));
+                }
+            });
+
+        destServer = app.listen(2000);
+
+        session                    = new Session();
+        session.getAuthCredentials = function () {
+            return null;
+        };
+        session.id                 = 'sessionId';
+
+        proxy = new Proxy(proxyHostname, proxyPort1, proxyPort2);
+        proxy.openSession('http://127.0.0.1:2000/', session);
+    });
+
+    after(function () {
+        destServer.close();
+        proxy.close();
+        requestAgent.resetKeepAliveConnections();
+    });
+
+    // Tests
+    describe('Pages', function () {
+        var testDocumentEncoding = function (originUrl, expectedCharset, expectedBody, done) {
+            var url     = getProxyUrl('http://127.0.0.1:2000' + originUrl);
+            var options = {
+                url:     url,
+                headers: {
+                    accept: 'text/html,application/xhtml+xml,application/xml;q=0.9,*!/!*;q=0.8'
+                }
+            };
+
+            request(options, function (err, res, body) {
+                compareCode(body, expectedBody);
+
+                done();
+            });
+        };
+        var noop                 = function () {
+        };
+        var getExpectedSrcStr    = function (src, charsetStr, addBOM) {
+            var charset           = {
+                get: function () {
+                    return charsetStr;
+                },
+
+                fromMeta: noop
+            };
+            var proxyResources    = {
```

Comment:
Add new line here, to correct formatting. Always separate object and array vars with the new line